File: solutions/n_factorful/sol.cpp
// Written by: wyattwismer

#include<bits/stdc++.h>
using namespace std;
#define N 1000007
#define M 11

int T,a,b,n,cnt;
int p[N];
int ps[M][N];
int main(){
    for(int i=2;i<N;i++)
        if(p[i] == 0) for(int j=i; j<N; j+=i) p[j]++;
    for(int i=0;i<M;i++)
        for(int j=1;j<N;j++) ps[i][j] = ps[i][j-1] + (p[j]==i);
    cin>>T;
    while(T--){
        cin>>a>>b>>n;
        cout << ps[n][b] - ps[n][a-1] << endl;
    }
}

File: solutions/n_factorful/sol.py
# Written by: wyattwismer

N  = 1000007
M  = 11
p  = [0]*N
ps = [[0]*N for i in xrange(M)] 

for i in xrange(2,N):
    if p[i] == 0:
        for j in xrange(i,N,i):
            p[j] += 1

for i in xrange(M):
    for j in xrange(1,N):
        ps[i][j] = ps[i][j-1] + int(p[j]==i)

T = int(raw_input())
for TT in xrange(T):
    a,b,n = map(int, raw_input().split())
    print ps[n][b] - ps[n][a-1]

